lib/prx_auth/resource_map.rb

```diff
@@ -1,13 +1,20 @@
 module PrxAuth
-  class ResourceMap
+  class ResourceMap < Hash
     WILDCARD_KEY = '*'
 
     def initialize(mapped_values)
+      super() do |hash, key|
```

Comment:
What madness be this?  (I'm not sure I follow what this is doing).

Reply:
I'll go through everything happening here since I am not sure exactly what you're asking about.

1. the parentheses after `super` are required or else it will automatically be treated as though I called `super(mapped_values)`
2. In this context, I am calling the superclass constructor `Hash.new do |hash, key|...end` which causes the passed block to be called for calls to `#[](key)` that are not in the hash.
3. In the block, I am returning the value for the wildcard when `map['*']` / `map.[]('*')` is called, but not actually storing that value in the hash itself so that e.g. it's not included in the `Enumerable` methods or `#keys`.

This could definitely be too clever by half, but I think that it aligns with how I think about these maps (the wildcard isn't really a part of the map, per se, but impacts how queries are handled). I could be convinced to store it directly in the hash instead of as a separate instance variable, but would still want to e.g. exclude it from `keys` etc

Reply:
Ahhh, okay.  I was getting hung up on (3).  Yeah, i think that all makes sense.

Storing wildcards outside the hash sound better to me.  You can more reliably treat it as a hash, and optionally ask it about wildcard permissions.  Which lines up with how we've done things so far.